packages/flutter/lib/src/material/data_table.dart

```diff
@@ -0,0 +1,757 @@
+        child: contents
+      );
+    }
+    return new TableCell(
+      verticalAlignment: TableCellVerticalAlignment.fill,
+      child: contents
+    );
+  }
+
+  Widget _buildHeadingCell({
+    EdgeInsets padding,
+    Widget label,
+    String tooltip,
+    bool numeric,
+    VoidCallback onSort,
+    bool sorted,
+    bool ascending
+  }) {
+    if (onSort != null) {
+      final Widget arrow = new _SortArrow(
+        visible: sorted,
+        down: sorted ? ascending : null,
+        duration: _kSortArrowAnimationDuration
+      );
+      final Widget arrowPadding = new SizedBox(width: _kSortArrowPadding);
+      label = new Row(
+        children: numeric ? <Widget>[ arrow, arrowPadding, label ]
+                          : <Widget>[ label, arrowPadding, arrow ]
+      );
+    }
+    label = new Container(
+      padding: padding,
+      height: _kHeadingRowHeight,
+      child: new Align(
+        alignment: new FractionalOffset(numeric ? 1.0 : 0.0, 0.5), // TODO(ianh): RTL for non-numeric
+        child: new AnimatedDefaultTextStyle(
+          style: new TextStyle(
+            // TODO(ianh): font family should be Roboto; see https://github.com/flutter/flutter/issues/3116
+            fontWeight: FontWeight.w500,
+            fontSize: 12.0,
+            color: onSort != null && sorted ? Colors.black87 : Colors.black54,
+            height: 56.0/12.0
+          ),
+          duration: _kSortArrowAnimationDuration,
+          child: label
+        )
+      )
+    );
+    if (tooltip != null) {
+      label = new Tooltip(
+        message: tooltip,
+        child: label
+      );
+    }
+    if (onSort != null) {
+      label = new InkWell(
+        onTap: onSort,
+        // TODO(ianh): When we do RTL, we need to use 'end' ordering for the non-numeric case
+        child: label
+      );
+    }
+    return label;
+  }
+
+  Widget _buildDataCell({
+    EdgeInsets padding,
+    Widget label,
+    bool numeric,
+    bool placeholder,
+    bool showEditIcon,
+    VoidCallback onTap,
+    VoidCallback onSelectChanged
+  }) {
+    if (showEditIcon) {
+      final Widget icon = new Icon(icon: Icons.edit, size: 18.0);
+      label = new Flexible(child: label);
+      label = new Row(children: numeric ? <Widget>[ icon, label ] : <Widget>[ label, icon ]);
+    }
+    label = new Container(
+      padding: padding,
+      height: _kDataRowHeight,
+      child: new Align(
+        alignment: new FractionalOffset(numeric ? 1.0 : 0.0, 0.5), // TODO(ianh): RTL for non-numeric
+        child: new DefaultTextStyle(
+          style: new TextStyle(
+            // TODO(ianh): font family should be Roboto; see https://github.com/flutter/flutter/issues/3116
+            fontSize: 13.0,
+            color: placeholder ? Colors.black38 : Colors.black87 // TODO(ianh): defer to theme, since this won't work in e.g. the dark theme
+          ),
+          child: new IconTheme(
+            data: new IconThemeData(
+              color: Colors.black54
```

Comment:
I'll file a bug on making the data tables support the dark theme. The spec doesn't say anything about it.